src/analysis/fourier/poisson_summation.lean

```diff
@@ -0,0 +1,226 @@
+/-
+Copyright (c) 2023 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import analysis.fourier.add_circle
+import analysis.fourier.fourier_transform
+import measure_theory.integral.exp_decay
+import analysis.special_functions.gaussian
+
+/-!
+# Poisson's summation formula
+
+We prove Poisson's summation formula `∑ (n : ℤ), f n = ∑ (n : ℤ), 𝓕 f n`, where `𝓕 f` is the
+Fourier transform of `f`, under the following hypotheses:
+* `f` is a continuous function `ℝ → ℂ`.
+* The sum `∑ (n : ℤ), 𝓕 f n` is convergent.
+* For all compacts `K ⊂ ℝ`, the sum `∑ (n : ℤ), sup { ‖f(x + n)‖ | x ∈ K }` is convergent.
+
+The last condition is automatic if we assume `f` has sufficiently rapid decay (e.g. if it is a
+Schwartz function); this will be added in future.
```

Comment:
Can you explicitly leave that as a TODO? 
```suggestion
Schwartz function).

## TODO

Prove that Schwartz functions satisfy the last condition
```
That way, if you don't get around to do it, someone else will be more likely to.

Reply:
I already have the code, actually -- at least, I can show that any function which is $O(|x|^{-b})$ at infinity for some $b > 1$ satisfies my hypotheses (which is a vastly weaker condition than Schwartz). The problem is that the API for Schwartz functions is not very intuitive, or at least I don't find it so.

Reply:
Ah okay  Still doesn't hurt to explicitly mark it as a TODO.